extensions/amp-video/1.0/video-iframe.js

```diff
@@ -0,0 +1,168 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../src/preact';
+import {Deferred} from '../../../src/utils/promise';
+import {forwardRef} from '../../../src/preact/compat';
+import {
+  useCallback,
+  useEffect,
+  useImperativeHandle,
+  useLayoutEffect,
+  useMemo,
+  useRef,
+} from '../../../src/preact';
+
+const DEFAULT_SANDBOX = [
+  'allow-scripts',
+  'allow-same-origin',
+  'allow-popups',
+  'allow-popups-to-escape-sandbox',
+  'allow-top-navigation-by-user-activation',
+].join(' ');
+
+/**
+ * @param {T} prop
+ * @return {current: ?T}
+ * @template T
+ */
+function usePropRef(prop) {
+  const ref = useRef(null);
+  useEffect(() => {
```

Comment:
No need for `useEffect`. The ref can be updated immediately. E.g.

```
const ref = useRef()
ref.current = prop;
return ref;
```

And I agree. It could be a pattern indeed. As long as it doesn't get overused.

I think @realPrimoh had a similar problem recently.